test/IntegrationTests/StackExchangeRedisTests.cs

```diff
@@ -29,38 +26,29 @@ namespace IntegrationTests;
 [Collection(RedisCollection.Name)]
 public class StackExchangeRedisTests : TestHelper
 {
-    private const string ServiceName = "TestApplication.StackExchangeRedis";
     private readonly RedisFixture _redis;
 
     public StackExchangeRedisTests(ITestOutputHelper output, RedisFixture redis)
         : base("StackExchangeRedis", output)
     {
         _redis = redis;
-        SetEnvironmentVariable("OTEL_SERVICE_NAME", ServiceName);
     }
 
     [Fact]
     [Trait("Category", "EndToEnd")]
     [Trait("Containers", "Linux")]
     public async Task SubmitsTraces()
     {
-        using var agent = await LegacyMockZipkinCollector.Start(Output);
-
-        RunTestApplication(agent.Port, arguments: $"--redis {_redis.Port}");
-
-        const int expectedSpansCount = 8;
-
-        var spans = await agent.WaitForSpansAsync(expectedSpansCount);
-
-        using (new AssertionScope())
+        using var collector = await MockSpansCollector.Start(Output);
+        const int spanCount = 8;
+        for (int i = 0; i < spanCount; i++)
```

Comment:
seems a candidate for later refactoring.

Reply:
I was also considering to remove the "span count" and just wait for the first span. But I had no strong feeling what would be better and I did not want to do too much in this PR.